compendium/DeclarativeServices/src/manager/ConfigurationNotifier.cpp

```diff
@@ -183,21 +186,29 @@ namespace cppmicroservices
             auto registry = mgr->GetRegistry();
             auto logger = mgr->GetLogger();
             auto configNotifier = mgr->GetConfigNotifier();
-            auto managers = mgr->GetManagers();
-
             try
             {
                 auto compManager = std::make_shared<ComponentManagerImpl>(newMetadata,
                                                                           registry,
                                                                           bundle.GetBundleContext(),
                                                                           logger,
                                                                           asyncWorkService,
-                                                                          configNotifier,
-                                                                          managers);
+                                                                          configNotifier);
                 if (registry->AddComponentManager(compManager))
                 {
-                    managers->push_back(compManager);
-                    compManager->Initialize();
+                    if (auto const& extension = extensionRegistry->Find(bundle.GetBundleId()); extension)
+                    {
+                        extension->AddComponentManager(compManager);
+                        compManager->Initialize();
+                    }
+                    else
+                    {
+                        logger->Log(cppmicroservices::logservice::SeverityLevel::LOG_ERROR,
+                                    "Failed to find ComponentManager with name " + newMetadata->name
+                                        + " from bundle with Id "
+                                        + std::to_string(bundleContext.GetBundle().GetBundleId()));
+                        throw std::exception();
```

Comment:
I don't understand why there is a `throw` here. This will cause execution to proceed to the `catch (std::exception const&)` block and then log a message which is misleading (i.e. ComponentManager creation did not fail). I think the `throw` should be removed.